### ISSUE TITLE
'FBReactNativeSpec_visionOS/FBReactNativeSpec_visionOS.h' file not found

### ISSUE THREAD

**Issue by billyjacoby** [callstack/react-native-visionos#106]
### Description

When trying to compile and run the example XR app, the process fails with the error `'FBReactNativeSpec_visionOS/FBReactNativeSpec_visionOS.h' file not found`

### Steps to reproduce

1. Init a new project as described in the readme
3. Run the build in XCode

### React Native Version

0.73.4

### Affected Platforms

Other (please specify)

### Output of `npx react-native info`

```text
System:
  OS: macOS 14.2.1
  CPU: (12) arm64 Apple M2 Max
  Memory: 1.01 GB / 32.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.5.0
    path: ~/.nvm/versions/node/v20.5.0/bin/node
  Yarn:
    version: 4.1.0
    path: ~/.nvm/versions/node/v20.5.0/bin/yarn
  npm:
    version: 9.8.0
    path: ~/.nvm/versions/node/v20.5.0/bin/npm
  Watchman:
    version: 2024.01.15.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.15.2
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.2
      - iOS 17.2
      - macOS 14.2
      - tvOS 17.2
      - visionOS 1.0
      - watchOS 10.2
  Android SDK: Not Found
IDEs:
  Android Studio: 2023.1 AI-231.9392.1.2311.11330709
  Xcode:
    version: 15.2/15C500b
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.10
    path: /usr/bin/javac
  Ruby:
    version: 2.6.10
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.73.4
    wanted: 0.73.4
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false
```

### Stacktrace or Logs

```text
`/Users/billyjacoby/Programming-Local/rnVisionLATEST/node_modules/@callstack/react-native-visionos/Libraries/XR/RCTXRModule.mm:3:9 'FBReactNativeSpec_visionOS/FBReactNativeSpec_visionOS.h' file not found
`
```

### Reproducer

https://github.com/billyjacoby/callstack-visionos-repro

**Comment by degeloper34** [comment]
Same issue +1

**Comment by okwasniewski** [comment]
Hey, 

Can you try running `git clean -xdf visionos` to remove build folders and cache from visionOS folder.

And then rerun pod install. You might also need to remove and install node_modules

Let me know if this fixes the issue

**Comment by billyjacoby** [comment]
> Hey,
> 
> Can you try running `git clean -xdf visionos` to remove build folders and cache from visionOS folder.
> 
> And then rerun pod install. You might also need to remove and install node_modules
> 
> Let me know if this fixes the issue

This doesn't fix it. I ran `git clean -xdf visionos` and `yarn cache clean` and deleted node_modules before trying again too.

**Comment by okwasniewski** [comment]
Can you try if it's reproducible in a freshly initialised project?

**Comment by billyjacoby** [comment]
> Can you try if it's reproducible in a freshly initialised project? 

Yeah I actually tried in a brand new project on 2 different machines before posting the issue.

**Comment by RobPruzan** [comment]
I'm also reproducing on a new project, can't downgrade and re-init either since I can't find what's being persisted

**Comment by MP185496** [comment]
Same for me

**Comment by JeffreyCastellano** [comment]
Hey folks, was just a code generation issue. Had the same problem... 'FBReactNativeSpec_visionOS/FBReactNativeSpec_visionOS.h' should be 'FBReactNativeSpec/FBReactNativeSpec.h'. The updated header already has the appropriate code updates. Doing that cleared the error for me and I can launch XR immersions.  
![Screenshot 2024-02-08 at 3 33 46 PM](https://github.com/callstack/react-native-visionos/assets/623535/972c9fe0-b551-404d-9e1f-e5f7075dd5ca)

**Comment by billyjacoby** [comment]
> Hey folks, was just a code generation issue. Had the same problem... 'FBReactNativeSpec_visionOS/FBReactNativeSpec_visionOS.h' should be 'FBReactNativeSpec/FBReactNativeSpec.h'. The updated header already has the appropriate code updates. Doing that cleared the error for me and I can launch XR immersions. ![Screenshot 2024-02-08 at 3 33 46 PM](https://private-user-images.githubusercontent.com/623535/303475834-972c9fe0-b551-404d-9e1f-e5f7075dd5ca.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc0MjU3NDgsIm5iZiI6MTcwNzQyNTQ0OCwicGF0aCI6Ii82MjM1MzUvMzAzNDc1ODM0LTk3MmM5ZmUwLWI1NTEtNDA0ZC05ZTFmLWU1ZjcwNzVkZDVjYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIwOFQyMDUwNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05Y2I3ZjcxOWJkOGYwYzYzMjAzNmIwY2Q4NDJmZjcwZjRkMDU2NzdmYTNkNzNlMTI1YjVjMDhkMDVlOGViZDFhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.iacwi38_CDAShK7F-yilZA4v3KdleMnkd5q5CMSERGo)

Confirmed, this worked for me too!

@JeffreyCastellano Did you have to create that Alert from SwiftUI? I'm trying to figure out if the current XR implementation only allows for immersive content from Swift. And if so you would mind sharing the code you've got here from the App.swift file? 

I'm super new to Swift and trying to dip my toes in for VisionOS dev specifically

**Comment by JeffreyCastellano** [comment]
Alert is coming from React Native...its worth noting they may change the way that file is coming in to better support react native since the other devices won't have that XR profile but at least temporarily this solved it for me.

**Comment by okwasniewski** [comment]
Hey! 

I've released new version `0.73.4-1` with the fix. Let me know if this solves the issue

**Comment by billyjacoby** [comment]
Confirmed that this fixes it!
